### PR TITLE
[PKO-5] Deploy Package Operator to Hosted Clusters

### PR DESCRIPTION
**Pull request by kostola**
### Summary

With this PR, Package Operator running in HyperShift mode inside Management Clusters now deploys another Package Operator instance inside each Hosted Cluster.

Major changes involved are:
* `package-operator-package` is now a multi-component package containing the standard package as root component plus two other components: `remote-phase` and `hosted-cluster`
* `remote-phase` component is literally the old `remote-phase-package` that got moved there. This avoids the need to link the `remote-phase-package` with the `package-operator-package`.
* `hosted-cluster` component is the new component that deploys PKO in hosted clusters.
* `HostedClusterController` now deploys both `remote-phase` and `hosted-cluster` packages in each hosted cluster by using the same `package-operator-package` image and specifying the component.

Additional changes need to make things work:
* Package templating context now contains the ID of the package image being processed inside the `.package.image` variable. Currently used in `config/packages/package-operator/package-operator-manager.Deployment.yaml.gotmpl`.
* `PKO_REMOTE_PHASE_PACKAGE_IMAGE` env var and related `--remote-phase-package-image` flags are replaced by `PKO_PACKAGE_OPERATOR_PACKAGE_IMAGE` and `--package-operator-package-image`.

### Change Type

**Breaking Change** only in the way the package containing HyperShift components is configured: using `PKO_PACKAGE_OPERATOR_PACKAGE_IMAGE` instead of `PKO_REMOTE_PHASE_PACKAGE_IMAGE`.

### Check List Before Merging

- [x] This PR passes all pre-commit hook validations.
- [x] This PR is fully tested and regression tests are included.
- [x] Relevant documentation has been updated.

### CODE REVIEW

**Comment by kostola** [comment]
Requires #959 to complete integration tests

File: .vscode/launch.json
Comment (eqrx) [inline .vscode/launch.json:1]: Is it intended to commit this? If so, lets call it smth more explicit than "Launch" since it specifically creates a hosted cluster. Maybe we could even add other targets to make it easier to develop from vscode

**Review by eqrx — CHANGES_REQUESTED** [review]

File: .vscode/launch.json
Comment (kostola) [reply .vscode/launch.json:1]: @erdii you added this file...I can't answer this comment because I don't use VSCode with Go

File: cmd/build/package.go
Comment (kostola) [reply cmd/build/package.go:30]: exactly...I didn't see any CI error anymore after changing this

File: .vscode/launch.json
Comment (eqrx) [reply .vscode/launch.json:1]: remove it please and i approve it:)

Comment (kostola) [reply .vscode/launch.json:1]: removed

File: config/packages/package-operator/.test-fixtures/openshift-with-proxy/package-operator-manager.Deployment.yaml
Comment (kostola) [reply config/packages/package-operator/.test-fixtures/openshift-with-proxy/package-operator-manager.Deployment.yaml]: fixed

File: install.yaml
Comment (kostola) [reply install.yaml:3744]: I'll keep `latest` here since it was there before

File: internal/adapters/package.go
Comment (kostola) [reply internal/adapters/package.go]: fixed

File: internal/controllers/hostedclusters/hostedcluster_controller.go
Comment (kostola) [reply internal/controllers/hostedclusters/hostedcluster_controller.go]: not needed anymore

File: cmd/build/test.go
Comment (kostola) [reply cmd/build/test.go:145]: I used the same pattern as the upper part of the test, let's keep as it is for now

File: internal/controllers/hostedclusters/hostedcluster_controller.go
Comment (kostola) [reply internal/controllers/hostedclusters/hostedcluster_controller.go]: not needed anymore

File: cmd/build/generate.go
Comment (kostola) [reply cmd/build/generate.go:362]: it's grouped in the `default` case now

**Review by eqrx — APPROVED** [review]
lgtm but please wait for the review by @thetechnick

**Review by thetechnick — APPROVED** [review]
lgtm as well :+1:

**Pull request merged**
